File: file-translation-demo/csharp/FileTranslationDemo/FileTranslateClient.cs
using common;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.IO;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;
using System.Web;

namespace FileTranslationDemo
{
    public class FileTranslateClient
    {        
        private string baseUrl;
        public FileTranslateClient(string baseUrl) 
        {
            this.baseUrl = baseUrl;
        }
        public async Task<string> TranslateAsync(string[] files, string contractId, FileTranslationOption option)
        {
            string url = baseUrl + "/file-translate";

            Dictionary<string, object> headers = HttpUtils.BuildHeaders(
                new RozettaApiUser { AccessKey= option.AccessKey, SecretKey=option.SecretKey}, url);

            Dictionary<string, string> body = new Dictionary<string, string>{
                    { "fieldId", option.FieldId.ToString() },
                    { "targetLangs", JsonConvert.SerializeObject(option.Langs) }
            };
            if (!string.IsNullOrEmpty(contractId)){
                body.Add("contractId", contractId);
            }

            Dictionary<string, string> fileDict = new Dictionary<string, string>();
            foreach(string file in files)
            {
                fileDict.Add("files", file);
            }   

            var content = await HttpUtils.PostFileAsync(url, headers, body, fileDict);

            var byteArray = await content.ReadAsByteArrayAsync();

            var responseString = Encoding.UTF8.GetString(byteArray, 0, byteArray.Length);

            var definition = new { status = "", data = new { translateId = ""} };            

            var serverResp = JsonConvert.DeserializeAnonymousType(responseString, definition);

            if (serverResp.status!=ResponseStatus.Success)
                throw new Exception("upload file failed");

            return serverResp.data.translateId;

        }

        public Translate[] GetAllHistories(RozettaApiUser rozettaApiUser)
        {
            Translate[] ret = { };
            return ret;
        }
        public async Task<Translate> GetOneHistoryAsync(RozettaApiUser rozettaApiUser, string translateId)
        {
            string url = baseUrl + "/translate-result/" + translateId;

            Dictionary<string, object> headers = HttpUtils.BuildHeaders(rozettaApiUser, url);

            var content = await HttpUtils.SendAsync(HttpMethod.Get, url, headers, null);

            var byteArray = await content.ReadAsByteArrayAsync();

            var responseString = Encoding.UTF8.GetString(byteArray, 0, byteArray.Length);

            var definition = new { status = "", data = new Translate { } };

            var serverResp = JsonConvert.DeserializeAnonymousType(responseString, definition);

            if (serverResp.status != ResponseStatus.Success)
                return null;

            return serverResp.data;
        }

        public async Task<bool> DownloadZipAsync(RozettaApiUser rozettaApiUser,string ids, string savePath)
        {            
            string url = baseUrl + "/downloads?ids=" + HttpUtility.UrlEncode(ids);

            string apiPath = HttpUtility.UrlDecode(new Uri(url).PathAndQuery);
            string nonce = DateTimeOffset.Now.ToUnixTimeMilliseconds().ToString();
            string signature = Utils.GetSignature(nonce, apiPath, rozettaApiUser.SecretKey);

            Dictionary<string, object> headers=new Dictionary<string, object>{
                    { "accessKey", rozettaApiUser.AccessKey },
                    { "nonce", nonce },
                    { "signature", signature }
            };            

            var content = await HttpUtils.SendAsync(HttpMethod.Get, url, headers, null);

            MemoryStream stream = new MemoryStream();

            await content.CopyToAsync(stream);

            Utils.UnzipStream(stream, savePath);

            return true;
        }        
    }
}

File: file-translation-demo/vb.net/FileTranslationDemo/FileTranslateClient.vb
Imports common.common

Namespace FileTranslationDemo
    Public Class FileTranslateClient
        Private baseUrl As String

        Public Sub New(ByVal baseUrl As String)
            Me.baseUrl = baseUrl
        End Sub

        Public Async Function TranslateAsync(ByVal files As String(), ByVal [option] As FileTranslationOption) As Task(Of String)
            Dim url As String = Me.baseUrl & "/file-translate"
            Dim headers As Dictionary(Of String, Object) = HttpUtils.BuildHeaders(New RozettaApiUser With {
                .AccessKey = [option].AccessKey,
                .SecretKey = [option].SecretKey
            }, url)
            Dim body As Dictionary(Of String, String) = New Dictionary(Of String, String) From {
                {"fieldId", [option].FieldId.ToString()},
                {"targetLangs", Newtonsoft.Json.JsonConvert.SerializeObject([option].Langs)}
            }
            If [option].ContractId IsNot Nothing Then
                body.Add("contractId", [option].ContractId)
            End If
            Dim fileDict As Dictionary(Of String, String) = New Dictionary(Of String, String)()

            For Each file As String In files
                fileDict.Add("files", file)
            Next

            Dim content = Await HttpUtils.PostFileAsync(url, headers, body, fileDict)
            Dim byteArray = Await content.ReadAsByteArrayAsync()
            Dim responseString = Text.Encoding.UTF8.GetString(byteArray, 0, byteArray.Length)
            Dim definition = New With {
                .status = "",
                .data = New With {
                    .translateId = ""
                }
            }
            Dim serverResp = Newtonsoft.Json.JsonConvert.DeserializeAnonymousType(responseString, definition)
            If Not Equals(serverResp.status, ResponseStatus.Success) Then Throw New Exception("upload file failed")
            Return serverResp.data.translateId
        End Function

        Public Function GetAllHistories(ByVal rozetaApiUser As RozettaApiUser) As Translate()
            Dim ret As Translate() = {}
            Return ret
        End Function

        Public Async Function GetOneHistoryAsync(ByVal rozettaApiUser As RozettaApiUser, ByVal translateId As String) As Task(Of Translate)
            Dim url As String = Me.baseUrl & "/translate-result/" & translateId
            Dim headers As Dictionary(Of String, Object) = HttpUtils.BuildHeaders(rozettaApiUser, url)
            Dim content = Await HttpUtils.SendAsync(Net.Http.HttpMethod.[Get], url, headers, DirectCast(Nothing, Dictionary(Of String, Object)))
            Dim byteArray = Await content.ReadAsByteArrayAsync()
            Dim responseString = Text.Encoding.UTF8.GetString(byteArray, 0, byteArray.Length)
            Dim definition = New With {
                .status = "",
                .data = New Translate
            }
            Dim serverResp = Newtonsoft.Json.JsonConvert.DeserializeAnonymousType(responseString, definition)

            Debug.WriteLine($"responseString:{responseString}")

            If Not Equals(serverResp.status, ResponseStatus.Success) Then
                Return DirectCast(Nothing, Translate)
            End If
            Return serverResp.data
        End Function

        Public Async Function DownloadZipAsync(ByVal rozettaApiUser As RozettaApiUser, ByVal ids As String, ByVal savePath As String) As Task(Of Boolean)
            Dim url As String = Me.baseUrl & "/downloads?ids=" & Web.HttpUtility.UrlEncode(ids)
            Dim apiPath As String = Web.HttpUtility.UrlDecode(New Uri(CStr((url))).PathAndQuery)
            Dim nonce As String = DateTimeOffset.Now.ToUnixTimeMilliseconds().ToString()
            Dim signature As String = Utils.GetSignature(nonce, apiPath, rozettaApiUser.SecretKey)
            Dim headers As Dictionary(Of String, Object) = New Dictionary(Of String, Object) From {
                {"accessKey", rozettaApiUser.AccessKey},
                {"nonce", nonce},
                {"signature", signature}
            }
            Dim content = Await HttpUtils.SendAsync(Net.Http.HttpMethod.[Get], url, headers, DirectCast(Nothing, Dictionary(Of String, Object)))
            Dim stream As IO.MemoryStream = New IO.MemoryStream()
            Await content.CopyToAsync(stream)
            Utils.UnzipStream(stream, savePath)
            Return True
        End Function
    End Class
End Namespace
